src/Identity/test/Identity.Test/RetryHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Identity.Test
+{
+    internal class RetryHandler : DelegatingHandler
+        {
+            private readonly ITestOutputHelper _output;
+            private readonly int _maxRetries;
+            private TimeSpan _waitIntervalBeforeRetry;
+
+            public RetryHandler(
+                HttpClientHandler httpClientHandler,
+                ITestOutputHelper output,
+                TimeSpan initialWaitTime,
+                int maxAttempts) : base(httpClientHandler)
+            {
+                _waitIntervalBeforeRetry = initialWaitTime;
+                _output = output;
+                _maxRetries = maxAttempts;
+            }
+
+            protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                HttpResponseMessage result = null;
+                var url = request.RequestUri;
+                var method = request.Method;
+
+                for (var i = 0; i < _maxRetries; i++)
+                {
+                    try
+                    {
+                        _output.WriteLine($"Sending request '{method} - {url}' {i+1} attempt.");
+                        result = await base.SendAsync(request, cancellationToken);
+                        if (result.IsSuccessStatusCode)
+                        {
+                            return result;
+                        }
+                        else
+                        {
+                            _output.WriteLine($"Request '{method} - {url}' failed with {result.StatusCode}.");
```

Comment:
Is it useful to include the message body here? If the app has the developer exception page set up, it should have the exception stack trace in plain text.

Reply:
The error that I've seen here is a network connection/failure to resolve a host. I'm actually going to add an extra attempt at the end so that the error gets surfaced.